code/modules/projectiles/guns/projectile.dm

```diff
@@ -286,3 +279,25 @@
 		playsound(M, empty_sound, 100, 1)
 		return 0
 	return ..()
+
+/obj/item/weapon/gun/projectile/proc/RemoveAttach(var/mob/user)
+	to_chat(user, "<span class='notice'>You unscrew [silenced] from [src].</span>")
+	user.put_in_hands(silenced)
+	silenced = 0
+	w_class = W_CLASS_SMALL
+	update_icon()
+
+/obj/item/weapon/gun/projectile/verb/RemoveAttachments()
+	set name = "Remove Attachments"
+	set category = "Object"
+	set src in range(0)
```

Comment:
```suggestion
	set src in usr
```
this will make the verb only show up if the item is in the hands of the user (or inside, I guess)